src/EnergyPlus/SwimmingPool.cc

```diff
@@ -1011,13 +1015,61 @@ void SwimmingPoolData::update(EnergyPlusData &state)
     Real64 WaterMassFlow = state.dataLoopNodes->Node(this->WaterInletNode).MassFlowRate; // water mass flow rate
     if (WaterMassFlow > 0.0) state.dataLoopNodes->Node(this->WaterOutletNode).Temp = this->PoolWaterTemp;
 }
+
 void SwimmingPoolData::oneTimeInit_new([[maybe_unused]] EnergyPlusData &state)
 {
 }
+
 void SwimmingPoolData::oneTimeInit([[maybe_unused]] EnergyPlusData &state)
 {
 }
 
+void SwimmingPoolData::report(EnergyPlusData &state)
+{
+    // SUBROUTINE INFORMATION:
+    //       AUTHOR         Rick Strand, Ho-Sung Kim
+    //       DATE WRITTEN   October 2014
+
+    // PURPOSE OF THIS SUBROUTINE:
+    // This subroutine simply produces output for the swimming pool model.
+
+    // SUBROUTINE PARAMETER DEFINITIONS:
+    static constexpr std::string_view RoutineName("SwimmingPoolData::report");
+    Real64 constexpr MinDensity = 1.0; // to avoid a divide by zero
+
+    int SurfNum = this->SurfacePtr; // surface number index
+
+    // First transfer the surface inside temperature data to the current pool water temperature
+    this->PoolWaterTemp = state.dataHeatBalSurf->SurfInsideTempHist(1)(SurfNum);
+
+    // Next calculate the amount of heating done by the plant loop
+    Real64 Cp = FluidProperties::GetSpecificHeatGlycol(state, "WATER", this->PoolWaterTemp, this->GlycolIndex,
+                                                       RoutineName); // specific heat of water
+    this->HeatPower = this->WaterMassFlowRate * Cp * (this->WaterInletTemp - this->PoolWaterTemp);
+
+    // Now the power consumption of miscellaneous equipment
+    Real64 Density = FluidProperties::GetDensityGlycol(state, "WATER", this->PoolWaterTemp, this->GlycolIndex,
+                                                       RoutineName); // density of water
+    if (Density > MinDensity) {
```

Comment:
Hmm.  I'm not sure GetDensityGlycol will ever give back zero density, but this check is fine I guess.